### ISSUE TITLE
Add a popup notification when someone calls you

### ISSUE THREAD

**Issue by CP-Vub** [mattermost/mattermost-plugin-calls#125]
Currently, in the new calls (beta) in v7.0, when someone calls you, you get the same notification as you would get with a normal chat message. This does not really grab the attention of the receiver, who might think it's just a chat message notification or just miss it entirely due to the brevity of the notification, leaving the caller waiting indefinitely in an empty call.

Could an option be added to enable e.g. a popup window or a continuously ringing sound when someone calls you? This would greatly improve the usability of the call functionality. Currently, I am often still resorting back to MS Teams or skype in order to reach a colleague since they missed the call notification in mattermost.

**Comment by streamer45** [comment]
Thanks @CP-Vub for the great feedback. 

We had this kind of functionality mentioned before internally and have considered adding a "ringing" feature for calls in direct message channels. Wondering if that would cover your needs or would you expect something to work across all channel types instead?

/cc @itao

**Comment by CP-Vub** [comment]
Thanks @streamer45 for the quick and positive reply! 

Adding a ringing feature for calls in direct message channels would definitely already solve like 95% of all my personal use-case scenario's.  For group channels, a call typically needs to be anyway communicated well in advance in order to align the agenda's and availability of the group members so it's way less common that I want to call an entire group channel without notice. I'm not saying it wouldn't be handy to have in group channels as well but priority-wise it's lower on the ladder for me personally.

**Comment by Madscientiste** [comment]
Hello, i agree with @CP-Vub 

Adding a ringing sound when someone calls you would just fix the issue for us, currently the default notification sound for when someone is calling is confusing .. 

i wouldn't mind digging into the plugin and doing a PR.

**Comment by streamer45** [comment]
Thanks @Madscientiste , would definitely love a contribution. Let me just bring this up with the team before we commit to any work. I'll share an update here.

**Comment by Madscientiste** [comment]
@streamer45 thanks for the quick reply ! 

just incase, i might need help in the beginning if thats possible .. i'm not very familiar with the source

**Comment by streamer45** [comment]
Of course. We'd be more than happy to support you through the whole process :)

**Comment by streamer45** [comment]
We discussed this today and agreed it makes sense to have an additional notification mechanism for calls in DM channels. Have created a ticket to keep track of this at https://mattermost.atlassian.net/browse/MM-46823.

As soon as we have a UX proposal we can proceed with an implementation. Will give an update when that happens.

**Comment by francislavoie** [comment]
Just want to give my 2 cents: I'd prefer if the "ringing" only happens in 1 on 1 calls only, and not on group chats. We've been using calls in channels sometimes when we need like 3 people in a call, but not all ~15+ people in the channel. It would be unfortunate if everyone in the channel was ringed when we open a call for that purpose. We can @ anyone who needs to be in the call in the channel at the same time and that's typically enough for us.

**Comment by streamer45** [comment]
Absolutely, at this point we are only planning to add this functionality to DMs (1-1s). To extend it to other channels we'd have to decouple a bit the call object such as, for example, only people invited to a call (e.g. through calendar integrations and such) would get the ping. But that's quite far ahead in the future at the moment.

**Comment by Madscientiste** [comment]
Just being curious,
Is there is any news regarding this request ? 🤔

**Comment by streamer45** [comment]
We are looking to tackle this functionality in the coming year.

**Comment by smarteertech** [comment]
Any updates? As a small team working remotely, we really need this to come out of Microsoft teams (Its no longer free).

**Comment by streamer45** [comment]
> Any updates? As a small team working remotely, we really need this to come out of Microsoft teams (Its no longer free).

We are currently planning a more traditional "ringing" feature for calls in DMs which hopfeully should be ready in the next couple of months.

**Comment by benschhold** [comment]
cant wait for this, i miss all the calls

**Comment by streamer45** [comment]
Leaving a quick update as I know quite a few people have been waiting on this. We just started working on UX design for this functionality and planning to work on an implementation in the upcoming weeks. Hopefully the next update will be the final one :)

**Comment by Madscientiste** [comment]
Very cool !

**Comment by kaksikanaa** [comment]
Hello, any news about this feature and when it will be available?
And have I understood correctly that the notification feature includes also a ringing tone + popup windows both?

**Comment by kaksikanaa** [comment]
Hello @streamer45 !
This is truly awesome. Mattermost is just fantastic tool and this feature would make the Call plugin actually useful for remote workers and teams. Have you made any progress with the implementation? We are now testing the Call plugin in our organization and this is something we still need quite desperately before we could implement it fully.

**Comment by streamer45** [comment]
@kaksikanaa Thanks for your trust. Designs for this are done, we are planning to begin implementing next week. We are looking at next month to ship this functionality.

**Comment by kaksikanaa** [comment]
Hello @streamer45, just checking (before our holiday season starts at the end of this month) -> does it look like the new feature will be available already this month? No worries if not, just checking the estimated schedule for our own planning.

**Comment by streamer45** [comment]
@kaksikanaa The earliest it can be in would be with 8.0 Mattermost release which is scheduled for July.

**Comment by myezcare-dev** [comment]
That would be awesome. This is the only thing holding us up to get rid of teams

**Comment by streamer45** [comment]
We are finally releasing this functionality (thanks @cpoile for the incredible contribution!). 

Please keep in mind it's an experimental feature which needs to be explicitly turned on from the plugin settings. 

![image](https://github.com/mattermost/mattermost-plugin-calls/assets/1832946/3ef4e68c-a81f-4015-a508-015c73fd6c55)

Also note that it comes with some limitations which will be addressed in the next version (e.g. mobile support).

Feel free to report any issues or leave feedback. Thank you all for your patience!

**Comment by CP-Vub** [comment]
[celebrate]     Cédric Peeters reacted to your message:
________________________________

**Comment by CP-Vub** [comment]
[0]     Cédric Peeters reacted to your message:
________________________________

**Comment by myezcare-dev** [comment]
Thank you we will start implementing this and if you need any help we will reach out to you guys.Thank you so much.Thanks Pallav SaxenaOn Jul 8, 2023, at 10:12 AM, CP ***@***.***> wrote:﻿
[celebrate]     Cédric Peeters reacted to your message:

________________________________

From: Claudio Costa ***@***.***>

Sent: Saturday, July 8, 2023 4:31:19 PM

To: mattermost/mattermost-plugin-calls ***@***.***>

Cc: Cédric Peeters ***@***.***>; Mention ***@***.***>

Subject: Re: [mattermost/mattermost-plugin-calls] Add a popup notification when someone calls you (Issue #125)





We are finally releasing this functionality (thanks @cpoile<https://github.com/cpoile> for the incredible contribution!).



Please keep in mind it's an experimental feature which needs to be explicitly turned on from the plugin settings.



[image]<https://user-images.githubusercontent.com/1832946/252020379-3ef4e68c-a81f-4015-a508-015c73fd6c55.png>



Also note that it comes with some limitations which will be addressed in the next version (e.g. mobile support).



Feel free to report any issues or leave feedback. Thank you all for your patience!

**Comment by ageekhere** [comment]
Was mobile support (android, IOS) added yet? 
Does a new issue/feature need to be created in https://github.com/mattermost/mattermost-mobile/issues or is it already being tracked?

**Comment by streamer45** [comment]
@cpoile Could you confirm the above?

**Comment by cpoile** [comment]
@ageekhere You are correct that there isn't a ringing sound on mobile (by design). When the app is in the background or closed, the notification feature on mobile adds a custom push notification: `username is inviting you to a call`. When the app is open, you are given a notification banner similar to the desktop. Hope that answers the question?

**Comment by ageekhere** [comment]
@cpoile Is it possible to also have an open to enable ringing on mobile devices as well?

**Comment by cpoile** [comment]
@ageekhere It's not on our roadmap at the moment but it's always possible to add it if there's enough demand.

**Comment by ageekhere** [comment]
@cpoile Well i will say that having the ringing on mobile devices would be a very cool feature to have. It is hard to find apps that allow you to ring other mobile devices while using your own servers.

**Comment by smarteertech** [comment]
We need this as well. If someone not in their computer they would miss the notification.

**Comment by ageekhere** [comment]
Here is the current uservoice for mobile ringing/notification , 
https://mattermost.uservoice.com/forums/306457-general/suggestions/45341932-add-a-popup-notification-when-someone-calls-you-u

The suggestion includes both a popup window notification (which was added) and a ringing sound when someone calls you"

**Comment by ageekhere** [comment]
Has there been any changes to the roadmap for ringing on mobile devices?
Should i create a new Feature request issue for mobile ringing or is this already on the radar?

**Comment by streamer45** [comment]
@ageekhere In-app ringing is being added in https://github.com/mattermost/mattermost-mobile/pull/7984, which should be available in a future release.

**Comment by roman-g** [comment]
Could it be made to ring as a real call, for example via CallKit on iOS, as it works in other messangers?

**Comment by lllDez** [comment]
> Could it be made to ring as a real call, for example via CallKit on iOS, as it works in other messangers?

Same question here. I'm having difficulties understanding the current ringtone notification system on Android for Mattermost calls. It seems that when I receive an incoming call, I only get a standard notification popup and its associated sound. The Mattermost ringtone only starts playing after I open this notification popup. This behavior completely defeats the purpose of a ringtone, which is to grab attention from other apps or when the user isn't actively using their phone. Is this the intended functionality, or could it be a bug?

**Comment by ageekhere** [comment]
This was the reason why i switched to Nextcloud with Nextcloud talk, with that i am finally able to group call both Android and iPhone phones from my PC and the phones rings like it should, none of this the app has to be active thing.
